Issue #575, Quantum to Windows Changed, tests pass

Rokiis opened the pull request:
* File names changed from Quantum to Windows
* Functions and object names changed from quantum to windows
* Imports adjusted to direct to changed filenames
yarn test -All pass
yarn test -u -All pass


Please review and provide feedback.

Thanks.

klahnakoski commented:
... and thank you!

Rokiis commented on src/nimbledroid/config.js:
```diff
@@ -85,7 +85,7 @@ const generateSitesSummary = (count, numSites) => [
 const FILTERED_OUT_SITES = [
   'http://m.spiegel.de',
-  'http://m.spiegel.de/netzwelt/web/firefox-quantum-so-schlaegt-sich-der-mozilla-browser-gegen-google-chrome-a-1178579.html',
+  'http://m.spiegel.de/netzwelt/web/firefox-windows-so-schlaegt-sich-der-mozilla-browser-gegen-google-chrome-a-1178579.html',
```
I've tested the link, they both redirect to the same URL.

Rokiis replied:
The link reverted back to quantum

Rokiis commented:
Just to be clear: Does it only need to change from Quantum to Windows on text visible on the webpage? None of the internal naming needs to be changed?

klahnakoski commented:
Sorry for the delay.  Yes, only the visible portions should change.  But, changing the filenames is good for the future.

klahnakoski commented:
... which is still unclear:

I will accept either just a visual change, or a refactoring like you provided.  The only change you must make is to fix the URL which got caught up in your automated search-and-replace.

Rokiis commented:
Reverted the link requested back to normal, sorry for the delay, I can only really get to do stuff while at work. Please review. Thanks. 

P.s. I've also changed snapshot test as it was expecting to be Quantum Displayed, Please review that as well.

Rokiis commented:
Can I have an update on review please?

klahnakoski commented:
@airimovici   Is it OK if we change the directory names too?

I think we accept this change, and make separate issues to revert, assuming there is ever a problem.

airimovici commented:
@Rokiis @klahnakoski I'm ok with the directory names changing.

There seem to be some other occurrences of `quantum` in `test` folder in the root of the project:

examples: 

-  `import Quantum from '../../src/quantum/Quantum';` in `test/quantum/index-64.test.jsx` ([here](https://github.com/Rokiis/firefox-health-dashboard/blob/Quantum-to-Windows/test/quantum/index-64.test.jsx))
-  `<Quantum
          location={{ pathname: '/quantum/64' }}
          match={{ params: { bits: 64 } }}
        />` in `test/quantum/index-64.test.jsx`
-  `import TargetStatus from '../../src/quantum/target-status';` in `test/quantum/target-status.test.jsx`

and also some more in `test/views` ([here](https://github.com/Rokiis/firefox-health-dashboard/tree/Quantum-to-Windows/test/views))

Rokiis commented:
@airimovici  Yeah, I went for changing the displays of quantum, havent re-factored anything. Can do it if you want me to. However, the CI tests are failing on Windows changes due to pointing to wrong file afterwards. Looks like that would need to be changed as well

klahnakoski commented:
I have not forgotten about this PR.

Rokiis commented:
Thanks. So nothing else needs to be done, correct?

klahnakoski commented:
@Rokiis that is correct.  I am expecting some merge conflicts with other PRs and some interesting corner cases: I must be available to fix them right away.  I have been working on another project, which is finishing up now, so I should be able to get back to this project, and its backlog, soon.
